### DISCUSSION TITLE
Writing keys from p12 encrypted with legacy encryption

### DISCUSSION THREAD

**Discussion by msetina** [OpenSC/OpenSC#3080] [Q&A] [answered]
I want to write keys and cert from an old p12 file. pkcs15-init returns:
error:0308010C:digital envelope routines::unsupported
Using openssl, I get a better description of the problem. It seams that it is encrypted in a depricated encryption. openssl indicates it needs to be used with -legacy argument. And it works.
Is there an equal option for pkcs15-init?

**Comment by msetina** [comment]
The encryption is RC2-40-CBC

**Comment by Jakuje** [comment, marked as answer]
I think you would have to load the legacy provider for openssl. Then the "cipher" should be available, but the RC2 is really a joke and I do not think this is something we want to support.

We load the legacy provider for some uses already:

https://github.com/OpenSC/OpenSC/blob/a94af7f6ca0c190cbe9eb92fc17d6814709693a9/src/libopensc/ctx.c#L838

So it might be enough just to use the right library context for the operation in the pkcs15-init

**Comment by msetina** [comment]
Thank you. I hope that when I collect the new version of p12, I do not get the same packaging.
